src/services/getEditsForFileRename.ts

```diff
@@ -1,65 +1,180 @@
+                const elements: ReadonlyArray<Expression> = isArrayLiteralExpression(property.initializer) ? property.initializer.elements : [property.initializer];
+                let foundExactMatch = false;
+                for (const element of elements) {
+                    foundExactMatch = tryUpdateString(element) || foundExactMatch;
+                }
+                if (!foundExactMatch && propertyName === "include") {
+                    const includes = mapDefined(elements, e => isStringLiteral(e) ? e.text : undefined);
+                    const matchers = getFileMatcherPatterns(configDir, /*excludes*/ [], includes, useCaseSensitiveFileNames, currentDirectory);
+                    // If there isn't some include for this, add a new one.
+                    if (!getRegexFromPattern(Debug.assertDefined(matchers.includeFilePattern), useCaseSensitiveFileNames).test(newFileOrDirPath)) {
+                        changeTracker.insertNodeAfter(configFile, last(elements), createStringLiteral(relativePath(newFileOrDirPath)));
+                    }
+                }
+            }
+            else if (propertyName === "paths") {
+                if (!isObjectLiteralExpression(property.initializer)) continue;
+                for (const pathsProperty of property.initializer.properties) {
+                    if (!isPropertyAssignment(pathsProperty) || !isArrayLiteralExpression(pathsProperty.initializer)) continue;
+                    for (const e of pathsProperty.initializer.elements) {
+                        tryUpdateString(e);
+                    }
+                }
+            }
+        }
+
+        function tryUpdateString(element: Expression): boolean {
+            if (!isStringLiteral(element)) return false;
+            const elementFileName = combinePathsSafe(configDir, element.text);
+
+            const updated = oldToNew(elementFileName);
+            if (updated !== undefined) {
+                changeTracker.replaceRangeWithText(configFile!, createStringRange(element, configFile!), relativePath(updated));
+                return true;
+            }
+            return false;
         }
-    }
 
-    interface ToUpdate {
-        readonly sourceFile: SourceFile;
-        readonly toUpdate: StringLiteralLike | FileReference;
+        function relativePath(path: string): string {
+            return getRelativePathFromDirectory(configDir, path, /*ignoreCase*/ !useCaseSensitiveFileNames);
+        }
     }
-    function isRef(toUpdate: StringLiteralLike | FileReference): toUpdate is FileReference {
-        return "fileName" in toUpdate;
+
+    function isPathsPropertyName(propertyName: string): boolean {
+        switch (propertyName) {
+            case "includes":
+            case "files":
+            case "include":
+            case "exclude":
+            case "baseUrl":
+            case "typeRoots":
+            case "mapRoot":
+            case "rootDir":
+            case "rootDirs":

```

Comment:
except files, includes and exclude rest of these are in "compilerOptions" property. So you will miss those?